Defs/Ammo/Advanced/PlasmaGasCartridge.xml

```diff
@@ -0,0 +1,469 @@
+      </thingDefs>
+    </fixedIngredientFilter>
+    <products>
+      <Ammo_SWPlasmaGasCartridge_AP>100</Ammo_SWPlasmaGasCartridge_AP>
+    </products>
+  </RecipeDef>
+
+
+  <RecipeDef ParentName="ChargeAmmoRecipeBase">
+    <defName>MakeAmmo_SWPlasmaGasCartridge_Ion</defName>
+    <label>make Plasma Gas Cartridges (Ion)  x100</label>
+    <description>Craft 100 SWPlasmaGas Cartridges (Ion) .</description>
+    <jobString>Making Plasma Gas Cartridges (Ion) .</jobString>
+    <ingredients>
+      <li>
+        <filter>
+          <thingDefs>
+            <li>Plasteel</li>
+          </thingDefs>
+        </filter>
+        <count>6</count>
+      </li>
+      <li>
+        <filter>
+          <thingDefs>
+            <li>ComponentIndustrial</li>
+          </thingDefs>
+        </filter>
+        <count>3</count>
+      </li>
+    </ingredients>
+    <fixedIngredientFilter>
+      <thingDefs>
+        <li>Plasteel</li>
+        <li>ComponentIndustrial</li>
+      </thingDefs>
+    </fixedIngredientFilter>
+    <products>
+      <Ammo_SWPlasmaGasCartridge_Ion>100</Ammo_SWPlasmaGasCartridge_Ion>
+    </products>
+  </RecipeDef>
+  
+  
+  <!-- Damage and Hediff Defs -->
+  
+  
+  <DamageDef>
+    <defName>PJ_BlasterDam</defName>
+    <label>Blaster bolt</label>
+    <workerClass>DamageWorker_AddInjury</workerClass>
+    <externalViolence>true</externalViolence>
+    <isExplosive>true</isExplosive>
+    <deathMessage>{0} has been blasted!</deathMessage>
+    <hediff>Burn</hediff>
+    <harmAllLayersUntilOutside>true</harmAllLayersUntilOutside>
+    <impactSoundType>Blunt</impactSoundType>
+    <armorCategory>Sharp</armorCategory>
+    <minDamageToFragment>15</minDamageToFragment>
+    <defaultDamage>3</defaultDamage>
+    <buildingDamageFactor>3</buildingDamageFactor>
+    <plantDamageFactor>3</plantDamageFactor>
+    <explosionAffectOutsidePartsOnly>false</explosionAffectOutsidePartsOnly>
+    <explosionHeatEnergyPerCell>5</explosionHeatEnergyPerCell>
+    <explosionCellMote>Mote_BlastFlame</explosionCellMote>
+    <explosionColorCenter>(1, 0.5, 0.3)</explosionColorCenter>
+    <explosionColorEdge>(0.6, 0.5, 0.4)</explosionColorEdge>
+    <soundExplosion>Explosion_Flame</soundExplosion>
+    <combatLogRules>Damage_Flame</combatLogRules>
+    <canUseDeflectMetalEffect>false</canUseDeflectMetalEffect>
+  </DamageDef>
+
+	<DamageDef>
+    <defName>SW_Stun_Damage</defName>
+    <label>stun</label>
+    <workerClass>DamageWorker_AddInjury</workerClass>
+    <externalViolence>true</externalViolence>
+    <deathMessage>{0} has been shocked to death.</deathMessage>
+    <hediff>Burn</hediff>
+    <hediffSolid>Burn</hediffSolid>
+    <harmAllLayersUntilOutside>true</harmAllLayersUntilOutside>
+    <impactSoundType>Slice</impactSoundType>
+    <armorCategory>Sharp</armorCategory>
+	<additionalHediffs>
+      <li>
+        <hediff>SW_Stun</hediff>
+      </li>
+    </additionalHediffs>
+  </DamageDef>
+		
+		
+
+  <HediffDef>

```

Comment:
Damage and Hediff defs should also be put in separate XML files in the Star Wars patch folder.

Reply:
Done.